data/pigui/modules/info-view/04-missions.lua

```diff
@@ -102,13 +102,27 @@ local function makeMissionRows()
 		end
 
 		local playerSystem = Game.system or Game.player:GetHyperspaceTarget()
-		local dist = playerSystem:DistanceTo(mission.location)
 		local days = math.max(0, (mission.due - Game.time) / (24*60*60))
 
+		-- Use AU for interplanetary, LY for interstellar distances
+		local dist, dist_display
+		if mission.location:IsSameSystem(playerSystem.path) then
+			if mission.location:IsBodyPath() then
+				local systemBody = mission.location:GetSystemBody().body
+				dist = Game.player:GetPositionRelTo(systemBody):length()
```

Comment:
This body is not a SystemBody, it's just a regular Body. Would strongly recommend the variable not be named systemBody as that usually implies the type of the variable is actually the SystemBody returned from GetSystemBody().

(Variable names are important in dynamically-typed languages...)